### PR TITLE
34 single place page

### PR DESCRIPTION
**Pull request by SajedaIsmail**
single page is styled and created  …

some edits on slider style
some edits on review style
Relates #34

### CODE REVIEW

File: client/src/App/index.js [hunk 1]
```diff
@@ -10,4 +10,4 @@ const App = () => (
   </div>
 );
 
-export default App;
+export default App;
```
Comment (Amoodaa) [inline client/src/App/index.js]: unrelated file, just revert the eof line

File: client/src/App/style.css [hunk 2]
```diff
@@ -1,8 +1,4 @@
 /* replace into the place page when created */
```
Comment (Amoodaa) [inline client/src/App/style.css]: remove the comment as well

File: client/src/pages/singlePlace/FakeData.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const singleSitSpot={
+  name: "Dkhhkj jlkhl", 
+  country: "UK", 
+  city: "London", 
+  website: "http://hkjhk.com",
+}
+
+export default singleSitSpot;
```
Comment (Amoodaa) [inline client/src/pages/singlePlace/FakeData.js]: is your eslint working?

File: client/src/pages/singlePlace/index.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    {
+      id: 1,
+    reviewee: { Name: 'Hanaa', Country: 'UK', Age: '49-51' },
+    reviewDate: new Date().toLocaleDateString(),
+    title: 'OK THAT WAS Coool as  sa asfas dasdsad',
+    magicalFactors: [
+      {
+        value: { text: 'Solo Traveller Friendly', imgSrc: 'family' },
+        reviewText:
+          'kasjlksajdlkjkasjlksajdlkjkasjlksajdlkjkasjlksajdlkjkasjlksajdlkjkasjlksajdlkjkasjlksajdlkjkasjlksajdlkjkasjlksajdlkjkasjlksajdlkjkasjlksajdlkjkasjlksajdlkjkasjlksajdlkjkasjlksajdlkjkasjlksajdlkj',
```
Comment (Amoodaa) [inline client/src/pages/singlePlace/index.js]: remove this from line 15-113
just keep the `reviews: []`

File: client/src/pages/singlePlace/index.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+            </span>
+          </Button></Link>
+        </div>
+        <p className="heading-name-location">{`${name}, ${country}, ${city}`} </p>
+        <a href={website} className="heading-website">{website}</a>
+        <div className="singlePlace-slider">
+        <ImageCarousel slides={images} smallTitle />
+        </div>
+        <span className="recommended-by-text">Recommended by {reviews.length} contributors</span>
+        <Link to="/review"><Fab /></Link>
```
Comment (Amoodaa) [inline client/src/pages/singlePlace/index.js]: eslint not working, format your files

File: client/src/pages/singlePlace/style.css [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+.heading-website {
+   font-size: 1.2rem;
+  color: var(--main-color);
+  font-weight: 670;
+  font-family: "Raleway", sans-serif;
+  margin: 0 2.5%;
+  margin-bottom: 2%;
+}
+
+.singlePlace-slider {
```
Comment (Amoodaa) [inline client/src/pages/singlePlace/style.css]: should be
```suggestion
.single-place__slider {
```

File: client/src/pages/singlePlace/style.css [hunk 7]
```diff
@@ -0,0 +1,88 @@
+.singlePlace-header-img {
```
Comment (Amoodaa) [inline client/src/pages/singlePlace/style.css]:
```suggestion
.single-place__header-img {
```

File: client/src/pages/singlePlace/index.js [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+        <div className="reviews-container">
+        {reviews.map(({id, reviewee, reviewDate, title, magicalFactors, tripDetails, reviewText, preservation})=>
+        <Review key={id} reviewee={reviewee} reviewDate={reviewDate} title={title} magicalFactors={magicalFactors} tripDetails={tripDetails} reviewText={reviewText} preservation={preservation} />
```
Comment (Amoodaa) [inline client/src/pages/singlePlace/index.js]: this isnt how i wrote props to the review,
from the https://github.com/GSG-G7/sitspot/blob/6c2c3b9fcaf928d0f390966ccb161136420b798f/client/src/components/Review/index.js#L172-L174
you should just give it whole review object, without destructuring it

File: client/src/pages/singlePlace/index.js [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+
+  render() {
+    const { name, country, city, website, images } = this.props;
+    const { reviews } = this.state;
+    return (
+      <>
+        <img
+          src={backgroundImage}
+          alt="background"
+          className="singlePlace-header-img"
```
Comment (Amoodaa) [inline client/src/pages/singlePlace/index.js]:
```suggestion
          className="single-place__header-img"
```

File: client/src/pages/singlePlace/index.js [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+          <Link to="/review"><Button onClick={() => {}} className="buttons-container__add">
+            <Icon type="plus" style={{ color: '#fff' }} />
+            <span className="buttons-container__add__text">
+              Add your recommendation
+            </span>
+          </Button></Link>
+        </div>
+        <p className="heading-name-location">{`${name}, ${country}, ${city}`} </p>
+        <a href={website} className="heading-website">{website}</a>
+        <div className="singlePlace-slider">
```
Comment (Amoodaa) [inline client/src/pages/singlePlace/index.js]:
```suggestion
        <div className="single-place__slider">
```
respectively css changes

**Comment by Amoodaa** [comment]
address react console errors,
you made a map, so use `key` to each one,
and please fix them, there are more sematic issues,
check out the other pages, how we all did the headings background, we didnt use `` as its hard to manipulate

File: client/src/pages/singlePlace/FakeData.js (hunk 3 above)
Comment (SajedaIsmail) [reply client/src/pages/singlePlace/FakeData.js]: No, it didn't work yesterday but today is working ^_^

**Comment by SajedaIsmail** [comment]
> address react console errors,
> you made a map, so use `key` to each one,
> and please fix them, there are more sematic issues,
> check out the other pages, how we all did the headings background, we didnt use `` as its hard to manipulate

There are no issues in my react console, I think I have a problem with chrome,
I actually set a key with map, but forgot it in the last edits,
I know how you set the background for the header in all pages and I do the same in the Home page, I will try to do the same in this page.

File: client/src/pages/singlePlace/index.js [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+        </div>
+        <span className="recommended-by-text">
+          Recommended by {reviews.length} contributors
+        </span>
+        <Link to="/review">
+          <Fab />
+        </Link>
+        <div className="reviews-container">
+          {reviews.map(reviewItem => (
+            <Review review={reviewItem} />
```
Comment (Amoodaa) [inline client/src/pages/singlePlace/index.js]:
```suggestion
            <Review review={reviewItem} key={reviewItem.id} />
```

File: client/src/pages/singlePlace/index.js [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+    );
+  }
+}
+
+SinglePlace.propTypes = {
+  name: propTypes.string.isRequired,
+  country: propTypes.string.isRequired,
+  city: propTypes.string.isRequired,
+  website: propTypes.string.isRequired,
+  images: propTypes.arrayOf(String).isRequired,
```
Comment (Amoodaa) [inline client/src/pages/singlePlace/index.js]: this propTypes?
```suggestion
  images: propTypes.arrayOf(propTypes.string).isRequired,
```

File: client/src/pages/singlePlace/style.css [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+  font-size: 1.2rem;
+  font-family: "Raleway", sans-serif;
+  color: gray;
+  margin: 0 2.5%;
+  font-weight: 500;
+}
+
+.reviews-container {
+  margin: 3% 2.5%;
+  border-top: 1px solid grey
```
Comment (Amoodaa) [inline client/src/pages/singlePlace/style.css]: cut this border top and paste it in the component Review div
